pkg/services/pluginsintegration/serviceregistration/serviceregistration.go

```diff
@@ -2,31 +2,40 @@ package serviceregistration
 
 import (
 	"context"
+	"errors"
 
 	"github.com/grafana/grafana/pkg/plugins/auth"
 	"github.com/grafana/grafana/pkg/plugins/plugindef"
 	"github.com/grafana/grafana/pkg/services/accesscontrol"
 	"github.com/grafana/grafana/pkg/services/extsvcauth"
+	"github.com/grafana/grafana/pkg/services/pluginsintegration/pluginsettings"
 )
 
 type Service struct {
-	os extsvcauth.ExternalServiceRegistry
+	reg         extsvcauth.ExternalServiceRegistry
+	settingsSvc pluginsettings.Service
 }
 
-func ProvideService(os extsvcauth.ExternalServiceRegistry) *Service {
+func ProvideService(reg extsvcauth.ExternalServiceRegistry, settingsSvc pluginsettings.Service) *Service {
 	s := &Service{
-		os: os,
+		reg:         reg,
+		settingsSvc: settingsSvc,
 	}
 	return s
 }
 
 // RegisterExternalService is a simplified wrapper around SaveExternalService for the plugin use case.
 func (s *Service) RegisterExternalService(ctx context.Context, svcName string, svc *plugindef.ExternalServiceRegistration) (*auth.ExternalService, error) {
+	settings, err := s.settingsSvc.GetPluginSettingByPluginID(ctx, &pluginsettings.GetByPluginIDArgs{PluginID: svcName})
```

Comment:
how intensive is this db query? will this impact the startup time of grafana itself?

Reply:
It's a `Get` query. It is IMO negligible compared to the cost of writing the service account, its role/permissions/assignment and its credentials.
Also we should consider that this runs only if a plugin requested a service account setup.

Reply:
This only works for `app` type plugins. `datasource` plugins should still be able to define an externalServiceRegistration (afaik) but those doesn't have settings. I guess you need to pass the plugin.Type here and only do this check if the type is `plugins.TypeApp`.